keys_storage.go

```diff
@@ -1,19 +1,30 @@
 package doubleratchet
 
+import (
+	"bytes"
+	"sort"
+)
+
 // KeysStorage is an interface of an abstract in-memory or persistent keys storage.
 type KeysStorage interface {
 	// Get returns a message key by the given key and message number.
 	Get(k Key, msgNum uint) (mk Key, ok bool, err error)
 
 	// Put saves the given mk under the specified key and msgNum.
-	Put(k Key, msgNum uint, mk Key) error
+	Put(sessionID []byte, k Key, msgNum uint, mk Key, keySeqNum uint) error
```

Comment:
keySeqNo is a monotonically increasing number of the key in a session ( first key is 1, then 2 ...).

I have changed SessionID to:
```
type SessionID = []byte
```
so it can still be treated as a []byte to avoid casting when storing in the db, is that ok?

Reply:
I would strongly suggest not doing this. Aliasing (`type T1 = T2`) is a feature introduced only for the refactoring huge codebases, and using to mitigate design inefficient is definitely falls under the "abusing aliases" bucket.
Let me review PR first and tell if I agree on adding a new type here or not.

Reply:
Okay, so alias is definitely not the right solution here, and I don't see why SessionID should be a separate type. There is nothing special about it, it's just slice of bytes, and it's meaning (id of the session) captured by parameter name. We don't rename type of every int parameter (i.e. `type Len int`), right?
The litmus test here is to have at least one method for the type, which is different from basic type. I.e. if `SessionID` doesn't have own methods – there is probably a little reason to be a separate type.

Reply:
My idea was that we can have `bytes.Equal` encapsulated as a method of `SessionID` (`func (s SessionID) Equal(other SessionID) bool`) because it is used pretty often. Sorry I did not explain it.

On the other hand, when I look now at your comment, we can treat `bytes.Equal()` as `+` operation, i.e. consider it a basic operation (because `bytes` is in stdlib and always available). So, I am neutral here. 

In case of `[]int` type, I would probably lean towards creating a type because there is no `ints.Equal()` as far as I know.

I agree that type aliasing was used here improperly, although, I don't agree that it's only for refactoring :P

Reply:
sounds good to me, @adambabik any objections if we go for []byte then?

Reply:
@cammellos to sum up, I believe using `[]byte` like in the original code is the best way. Sorry for the confusion.

Reply:
> My idea was that we can have bytes.Equal encapsulated as a method of SessionID (func (s SessionID) Equal(other SessionID) bool) because it is used pretty often. 

Yeah, in that case, it would have more sense, but let's keep things explicit for now.

> I don't agree that it's only for refactoring :P

Well, that was the only motivation of adding them into the language, and the biggest fear was that people will start use them for other purposes, doing shortcuts in design :)